### PR TITLE
Feature/videofp

### PR DESCRIPTION
**Pull request by Andy--S**
In preparation for 'Canvas' and cauldron audio extension

### CODE REVIEW

File: src/main/java/com/red5pro/canvas/Red5ProCanvas.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package com.red5pro.canvas;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.red5.server.api.scope.IScope;
+
+public interface Red5ProCanvas {
```
Comment (Andy--S) [inline src/main/java/com/red5pro/canvas/Red5ProCanvas.java:8]: API object fills live stream with dynamic content

File: src/main/java/com/red5pro/canvas/DefaultFrameProvider.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.red5pro.canvas;
+
+public class DefaultFrameProvider implements FrameProvider {
+	/**
+	 * Builder for canvas api
+	 */
+	public static CanvasFactory factory;
```
Comment (Andy--S) [inline src/main/java/com/red5pro/canvas/DefaultFrameProvider.java:7]: Implement FrameProvider and call the factory for a canvas instance. Configure the canvas and roll

File: src/main/java/com/red5pro/canvas/Red5ProCanvas.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	 * Start output engine, option 1 local publish. Call after starting encoding
+	 * engine.
+	 * 
+	 * @param scope
+	 * @param name
+	 * @param record
+	 * @param appeand
+	 * @throws IOException
+	 */
+	void loopBack(IScope scope, String name, boolean record, boolean appeand) throws IOException;
```
Comment (Andy--S) [inline src/main/java/com/red5pro/canvas/Red5ProCanvas.java:46]: Use it in a standalone server loop back,  or forward it to another.

File: src/main/c++/video_processor2.h [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	PROCESS_RETURN_IMAGE,
+	PROCESS_RETURN_DATA
+};
+
+
+/*
+Required library export definitions.
+These are the typedefs Red5Pro uses to load video process modules.
+*/
+
```
Comment (mondain) [inline src/main/c++/video_processor2.h:78]: Are the sections below 79... meant as examples? if not I'd remove the comments before this goes public.

**Review by mondain — APPROVED** [review]
Looks good, just the one item for the h file.

File: src/main/c++/video_processor2.h (hunk 4 above)
Comment (Andy--S) [reply src/main/c++/video_processor2.h:78]: yup. For those unfamiliar with linking methods out of a compiled library.

**Pull request closed without merging**